rockcraft/models/project.py

```diff
@@ -179,15 +187,14 @@ class NameStr(pydantic.ConstrainedStr):
     regex = re.compile(NAME_REGEX)
 
 
-class Project(YamlModel):
+class Project(YamlModelMixin, BaseProject):
     """Rockcraft project definition."""
 
-    name: NameStr
-    title: Optional[str]
-    summary: str
+    name: NameStr  # type: ignore
+    # summary is Optional[str] in BaseProject
```

Comment:
we should make it mandatory in BaseProject

Reply:
I might as well fix this now, what else should be mandatory?
These are the fields:

```
    name: ProjectName
    title: Optional[ProjectTitle]
    version: VersionStr
    summary: Optional[SummaryStr]
    description: Optional[str]

    base: Optional[Any]

    contact: Optional[Union[str, UniqueStrList]]
    issues: Optional[Union[str, UniqueStrList]]
    source_code: Optional[AnyUrl]
    license: Optional[str]

    parts: Dict[str, Dict[str, Any]]  # parts are handled by craft-parts
```

Reply:
description too

Reply:
https://github.com/canonical/craft-application/issues/80